### PR TITLE
Create a poc processor

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- Map diligen response to court details [FLA-937](https://justice.gov.bc.ca/jira/browse/FLA-937)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java [hunk 1]
```diff
@@ -110,6 +111,21 @@ public DiligenDocumentDetails getDocumentDetails(BigDecimal documentId) {
 
     }
 
+    @Override
+    public ProjectFieldsResponse getDocumentFieldResponse(BigDecimal documentId) {
+
+        documentsApi.getApiClient().setBearerToken(diligenAuthService.getDiligenJWT(diligenProperties.getUsername(), diligenProperties.getPassword()));
+
+        try {
+
```
Comment (akroon3r) [inline src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java:120]:
```suggestion
```

File: src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java [hunk 2]
```diff
@@ -110,6 +111,21 @@ public DiligenDocumentDetails getDocumentDetails(BigDecimal documentId) {
 
     }
 
+    @Override
+    public ProjectFieldsResponse getDocumentFieldResponse(BigDecimal documentId) {
+
```
Comment (akroon3r) [inline src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenServiceImpl.java:116]:
```suggestion
```

File: src/backend/libs/efiling-diligen-client/src/test/java/ca/bc/gov/open/efilingdiligenclient/diligen/processor/FieldProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        List<Field> response = mapper.readValue(new String(
+                Files.readAllBytes(path2)), new TypeReference<List<Field>>(){});
+
+        ObjectNode actual = sut.getJson(new String(
+                Files.readAllBytes(path)), response);
+
+      //  System.out.println(actual);
```
Comment (akroon3r) [inline src/backend/libs/efiling-diligen-client/src/test/java/ca/bc/gov/open/efilingdiligenclient/diligen/processor/FieldProcessorTest.java:45]:
```suggestion
```

File: src/backend/libs/efiling-diligen-client/src/test/java/ca/bc/gov/open/efilingdiligenclient/diligen/processor/FieldProcessorTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    }
+
+
+}
```
Comment (akroon3r) [inline src/backend/libs/efiling-diligen-client/src/test/java/ca/bc/gov/open/efilingdiligenclient/diligen/processor/FieldProcessorTest.java:50]:
```suggestion
}

```

**Review by akroon3r — COMMENTED** [review]
two small things

**Comment by akroon3r** [comment]
Also what are java conventions for whitespace?

**Review by akroon3r — DISMISSED** [review]
😠

**Comment by alexjoybc** [comment]
> Also what are java conventions for whitespace?

between statements ?

**Comment by akroon3r** [comment]
> > Also what are java conventions for whitespace?
> 
> between statements ?

In general, seems like everyone does it differently.

File: src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/document/DocumentsApiDelegateImpl.java [hunk 5]
```diff
@@ -105,10 +125,26 @@ public DocumentsApiDelegateImpl(DiligenService diligenService,
...
+
+        try {
+
+            File resource = ResourceUtils.getFile("classpath:courtDetails.schema.json");
+            return fieldProcessor.getJson(new String(Files.readAllBytes(Paths.get(resource.getPath()))),
+                    response.getData().getFields());
+
+        } catch (IOException e) {
+
+            throw new RuntimeException(e);
```
Comment (TayGov) [inline src/backend/efiling-reviewer-api/src/main/java/ca/bc/gov/open/jag/efilingreviewerapi/document/DocumentsApiDelegateImpl.java:146]: We will need the dedicated exception at some point

File: src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenService.java [hunk 6]
```diff
@@ -10,4 +11,6 @@
 
     DiligenDocumentDetails getDocumentDetails(BigDecimal documentId);
 
+    ProjectFieldsResponse getDocumentFieldResponse(BigDecimal documentId);
```
Comment (TayGov) [inline src/backend/libs/efiling-diligen-client/src/main/java/ca/bc/gov/open/efilingdiligenclient/diligen/DiligenService.java:14]: This data was present in DiligenDocumentDetails.

**Review by TayGov — APPROVED** [review]
Approved to get it in. But two comments to consider.

**Pull request merged**